authors/apps/authentication/tests/test_authentication.py

```diff
@@ -254,3 +254,24 @@ def test_user_already_exists(self):
         self.assertTrue(response2.data['errors'])
         self.assertTrue(response2.data['errors']['username'])
         self.assertTrue(response2.data['errors']['email'])
+
+    def test_signup_nonalphanumeric_password(self):
+        """
+        test for request with a  non alphanumeric password
+        """
+        response = self.client.post(
+            reverse('authentication:user-signup'),
+            data={
+                "user": {
+                    "email": "tester3@mail.com",
+                    "username": "tester3",
+                    "password": "testetthhhhhkk"
+                }
+            },
+            format="json"
+        )
+        output = json.loads(response.content)
+        self.assertIn('Please ensure your password contains at least one letter and one numeral', str(output))
```

Comment:
What was the thinking behind the output response? Because ideally the password field should not be returned in the response body

Reply:
> What was the thinking behind the output response? Because ideally the password field should not be returned in the response body
 The password is not being returned in the response body.  All I'm getting from the output is the response error message so that I could be able to test it.  Also just to make clear this is a test file and not a view file.